src/Http/Http.Extensions/gen/StaticRouteHandlerModel/EndpointResponse.cs

```diff
@@ -27,18 +27,20 @@ internal class EndpointResponse
     internal EndpointResponse(IMethodSymbol method, WellKnownTypes wellKnownTypes)
     {
         WellKnownTypes = wellKnownTypes;
-        ResponseType = UnwrapResponseType(method);
+        ResponseType = UnwrapResponseType(method, out bool isAwaitable, out bool awaitableIsVoid);
         WrappedResponseType = method.ReturnType.ToDisplayString(EmitterConstants.DisplayFormat);
-        IsAwaitable = GetIsAwaitable(method);
-        IsVoid = method.ReturnsVoid;
+        IsAwaitable = isAwaitable;
+        IsVoid = method.ReturnsVoid || awaitableIsVoid;
```

Comment:
Personally, I find it a little surprising that `IsVoid` is true for `Task`.  The consuming code below also seems to be contorting a little to handle it.  I wonder if exposing a separate, mutually-exclusive `IsVoidAwaitable` would be clearer?  (If so, we might also want to rename the out param below.)

Reply:
The special-casing that we have to do when we create the delegate signatures is the only place where we have the awkward `(endpoint.Response.IsVoid && !endpoint.Response.IsAwaitable)` clause at the moment. If we did have a mutually exclusive property, we would have to do checks like `(endpoint.Response.IsAwaitable || endpoint.Response.IsAwaitableVoid)` in more places like [here](https://github.com/dotnet/aspnetcore/blob/7a4829de7a0995b1ccffabc8c8567a0193ebc669/src/Http/Http.Extensions/gen/StaticRouteHandlerModel/StaticRouteHandlerModel.Emitter.cs#L72), [here](https://github.com/dotnet/aspnetcore/blob/7a4829de7a0995b1ccffabc8c8567a0193ebc669/src/Http/Http.Extensions/gen/StaticRouteHandlerModel/StaticRouteHandlerModel.Emitter.cs#L98), and [here](https://github.com/dotnet/aspnetcore/blob/7a4829de7a0995b1ccffabc8c8567a0193ebc669/src/Http/Http.Extensions/gen/StaticRouteHandlerModel/StaticRouteHandlerModel.Emitter.cs#L84).

I think it might be better to rename this to `HasResponse` to indicate that it means the handler doesn't write anything to the `HttpContext`.